frontend/user/configProfile/configProfileController.js

```diff
@@ -1,37 +1,57 @@
+        'UserService', 'ImageService', '$rootScope', 'SCREEN_SIZES', 'MessageService', '$mdDialog', 'ObserverRecorderService',
+        ConfigProfileController
+    ]);
+    
+    function ConfigProfileController($state, STATES, $stateParams, ProfileService, CropImageService, AuthService, 
+        UserService, ImageService, $rootScope, SCREEN_SIZES, MessageService, $mdDialog, ObserverRecorderService) {
+
+        const configProfileCtrl = this;
 
         // Variable used to observe the changes on the user model.
-        var observer;
-
-        configProfileCtrl.user = AuthService.getCurrentUser();
-        configProfileCtrl.newUser = _.cloneDeep(configProfileCtrl.user);
-        configProfileCtrl.loading = false;
+        let observer;
         configProfileCtrl.cpfRegex = /^\d{3}\.\d{3}\.\d{3}\-\d{2}$/;
-        configProfileCtrl.photo_url = configProfileCtrl.newUser.photo_url;
         configProfileCtrl.loadingSubmission = false;
 
-        var HAS_ONLY_ONE_INSTITUTION_MSG = "Esta é a única instituição ao qual você é vinculado." +
-            " Ao remover o vínculo você não poderá mais acessar o sistema," +
-            " exceto por meio de novo convite. Deseja remover?";
+        const DELETE_ACCOUNT_ALERT = "Ao excluir sua conta você não poderá mais acessar o sistema," +
+            "exceto por meio de novo convite. Deseja realmente excluir sua conta?";
 
-        var HAS_MORE_THAN_ONE_INSTITUTION_MSG = "Ao remover o vínculo com esta instituição," +
-            " você deixará de ser membro" +
-            " e não poderá mais publicar na mesma," +
-            " no entanto seus posts existentes serão mantidos. Deseja remover?";
+        configProfileCtrl.$onInit = () => {
+            configProfileCtrl._setupUser();
+        }
 
-        var DELETE_ACCOUNT_ALERT = "Ao excluir sua conta você não poderá mais acessar o sistema," +
-            "exceto por meio de novo convite. Deseja realmente excluir sua conta?";
+        configProfileCtrl._setupUser = () => {
+            if(configProfileCtrl.canEdit()) {
+                configProfileCtrl.user = AuthService.getCurrentUser();
+                configProfileCtrl.newUser = _.cloneDeep(configProfileCtrl.user);
+                observer = ObserverRecorderService.register(configProfileCtrl.user);
+                configProfileCtrl._checkUserName();
+            } else {
+                UserService.getUser($stateParams.userKey)
+                    .then(user => configProfileCtrl.user = user);
+            }
+        }
+        
+        configProfileCtrl._checkUserName = () => {
+            if (configProfileCtrl.user.name === 'Unknown') {
+                delete configProfileCtrl.user.name;
+                delete configProfileCtrl.newUser.name;
+            }
+        }
+
+        configProfileCtrl.getPhoto = () => {
+            const user = configProfileCtrl.canEdit() ? configProfileCtrl.newUser : configProfileCtrl.user;
+            return user && user.photo_url;
```

Comment:
I think that using just js instead of lodash is much more efficient. See the [implementation](https://gist.github.com/harish2704/d0ee530e6ee75bad6fd30c98e5ad9dab) of loadsh get function